### DISCUSSION TITLE
Postgresql setup for local deployment

### DISCUSSION THREAD

**Discussion by SalomonHenke** [microsoft/aspire#3425] [Q&A] [answered]
Hey all

Im trying to use the postgresql component, but can't seem to get it to work. Help is appreciated. Tried reading the documentation, but unclear. I ran the add-migration but obviously not the update-database. What i've tried. 

Added the Aspire Postgresql Entity Framework component.  
![image](https://github.com/dotnet/aspire/assets/96744651/4272f5c6-3115-47dc-be74-2de86e990661)

Added the configuration to the Apphost:
![image](https://github.com/dotnet/aspire/assets/96744651/1c15b67e-f580-419d-a5b2-e0fa566dc7a8)

Added the reference to the Candidate.Api program.cs
![image](https://github.com/dotnet/aspire/assets/96744651/76561a0b-3c62-42af-8e8d-72b2fb2a7a81)

Added a CandidateContext:
![image](https://github.com/dotnet/aspire/assets/96744651/2d12f888-a1c2-4484-93f6-48df0dabb045)

Added a Handler where i use the Context:
![image](https://github.com/dotnet/aspire/assets/96744651/0d92e81f-d781-4f5b-9b35-d25b91112985)

Ressources view shows me that it initializes the db:
![image](https://github.com/dotnet/aspire/assets/96744651/ccdd4341-fdd0-4d50-acca-53e445d84507)

Migrations folder:
![image](https://github.com/dotnet/aspire/assets/96744651/8bd4083c-9ce2-44c4-ac1c-403529bd9767)

Error: 
![image](https://github.com/dotnet/aspire/assets/96744651/26dba090-c217-4b1f-89cd-3d6fe2f8354d)

**Comment by davidfowl** [comment, marked as answer]
You need to use entity framework to create the database, it’s not created for you automatically

**Reply by SalomonHenke** [reply]
Not sure i get it. The database is created via. the Aspire configuration? So i need to run my migration on it after it is created (e.g.                 _dbContext.Database.Migrate();
) or do i literally have to "create" it via. code?

**Reply by davidfowl** [reply]
> Not sure i get it. The database is created via. the Aspire configuration?

It isn't. When you call AddDatabase in the apphost it doesn't create the database, it just creates a connection string with a reference to the database.

>  So i need to run my migration on it after it is created (e.g. _dbContext.Database.Migrate();) or do i literally have to "create" it via. code?

There are multiple ways of creating the database from using [database init scripts in the container](https://github.com/dotnet/aspire-samples/tree/main/samples/DatabaseContainers), to using [EF migrations](https://github.com/dotnet/aspire-samples/tree/main/samples/DatabaseMigrations)
